src/core/multifeaturelistmodel.cpp

```diff
@@ -50,7 +50,26 @@ void MultiFeatureListModel::setFeatures( QgsVectorLayer *vl, const QString &filt
   }
   if ( !extent.isEmpty() )
   {
-    request.setFilterRect( extent );
+    QgsRectangle filterExtent = extent;
+    if ( QgsProject::instance()->crs() != vl->crs() )
+    {
+      QgsCoordinateTransform transform( QgsProject::instance()->crs(), vl->crs(), QgsProject::instance()->transformContext() );
+      try
+      {
+        filterExtent = transform.transform( extent );
+      }
+      catch ( const QgsException &e )
+      {
+        Q_UNUSED( e )
```

Comment:
Should we somehow show a message here?

Reply:
I don't think it's necessary. If we fail to reproject canvas extent, it means we'd have failed to render anything on the canvas to begin with. It's safe to be silent here.